Tests/IntegrationTests/PropagationTests.hpp

```diff
@@ -0,0 +1,477 @@
+
+  // ensure initial parameters are valid
+  CHECK_CLOSE_ABS(params.position(), pos, 0.125_um);
+  CHECK_CLOSE_ABS(params.time(), time, 1_ps);
+  CHECK_CLOSE_ABS(params.momentum(), mom, 0.125_eV);
+  // charge should be identical not just similar
+  BOOST_CHECK_EQUAL(params.charge(), charge);
+
+  return params;
+}
+
+/// Construct (initial) curvilinear parameters with covariance.
+inline Acts::CurvilinearParameters makeParametersCurvilinearWithCovariance(
+    double phi, double theta, double absMom, double charge) {
+  using namespace Acts;
+  using namespace Acts::UnitLiterals;
+
+  BoundVector stddev = BoundVector::Zero();
+  // TODO use momentum-dependent resolutions
+  stddev[eBoundLoc0] = 15_um;
+  stddev[eBoundLoc1] = 80_um;
+  stddev[eBoundTime] = 25_ns;
+  stddev[eBoundPhi] = 1_degree;
+  stddev[eBoundTheta] = 1.5_degree;
+  stddev[eBoundQOverP] = 1_e / 10_GeV;
+  BoundSymMatrix corr = BoundSymMatrix::Identity();
+  corr(eBoundLoc0, eBoundLoc1) = corr(eBoundLoc1, eBoundLoc0) = 0.125;
+  corr(eBoundLoc0, eBoundPhi) = corr(eBoundPhi, eBoundLoc0) = 0.25;
+  corr(eBoundLoc1, eBoundTheta) = corr(eBoundTheta, eBoundLoc1) = -0.25;
+  corr(eBoundTime, eBoundQOverP) = corr(eBoundQOverP, eBoundTime) = 0.125;
+  corr(eBoundPhi, eBoundTheta) = corr(eBoundTheta, eBoundPhi) = -0.25;
+  corr(eBoundPhi, eBoundQOverP) = corr(eBoundPhi, eBoundQOverP) = -0.125;
+  corr(eBoundTheta, eBoundQOverP) = corr(eBoundTheta, eBoundQOverP) = 0.5;
+  BoundSymMatrix cov = stddev.asDiagonal() * corr * stddev.asDiagonal();
+
+  auto withoutCov = makeParametersCurvilinear(phi, theta, absMom, charge);
+  return CurvilinearParameters(std::move(cov), withoutCov.position(),
+                               withoutCov.momentum(), withoutCov.charge(),
+                               withoutCov.time());
+}
+
+/// Construct (initial) neutral curvilinear parameters.
+inline Acts::NeutralCurvilinearTrackParameters makeParametersCurvilinearNeutral(
+    double phi, double theta, double absMom) {
+  using namespace Acts;
+  using namespace Acts::UnitLiterals;
+
+  // phi is ill-defined in forward/backward tracks. normalize the value to
+  // ensure parameter comparisons give correct answers.
+  if (not((0 < theta) and (theta < M_PI))) {
+    phi = 0;
+  }
+
+  Vector3D pos = Vector3D::Zero();
+  double time = 0.0;
+  Vector3D mom = absMom * makeDirectionUnitFromPhiTheta(phi, theta);
+  NeutralCurvilinearTrackParameters params(std::nullopt, pos, mom, time);
+
+  // ensure initial parameters are valid
+  CHECK_CLOSE_ABS(params.position(), pos, 0.125_um);
+  CHECK_CLOSE_ABS(params.time(), time, 1_ps);
+  CHECK_CLOSE_ABS(params.momentum(), mom, 0.125_eV);
+  // charge should be identical not just similar
+  BOOST_CHECK_EQUAL(params.charge(), 0);
+
+  return params;
+}
+
+// helpers to compare track parameters
+
+/// Check that two parameters object are consistent within the tolerances.
+///
+/// \warning Does not check that they are defined on the same surface.
```

Comment:
I do not think I can. E.g. if I compare two curvilinear parameters they might have exactly the same parameters, global positions, etc., but not be defined on an identical surface because they originate from two separate propagation calls.

Reply:
Right, that could actually happen